boards/arm/nucleo_l552ze_q/nucleo_l552ze_q_ns_defconfig

```diff
@@ -50,5 +50,5 @@ CONFIG_CORTEX_M_SYSTICK=y
 CONFIG_RUNTIME_NMI=y
 CONFIG_TRUSTED_EXECUTION_NONSECURE=y
 # Flash configuration for TFM
-CONFIG_FLASH_LOAD_OFFSET=0x51400
-CONFIG_FLASH_LOAD_SIZE=0x2E000
+CONFIG_FLASH_LOAD_OFFSET=0x51000
```

Comment:
@erwango in my opinion this is not a clean solution: we should not use the FLASH_LOAD_OFFSET to indicate the start of the non-secure Zephyr image. We should use the "non-secure" image partition and employ DT to extract the partition information. This is how it is done for MPS2, Musca and nRF platforms.

Reply:
This is right. I'll do that, but will probably end up in V2.5.0 if not blocking.

Reply:
@ioannisg Are you OK to merge this in for 2.4 as is if there isn't the bandwith to update quickly? https://github.com/zephyrproject-rtos/zephyr/pull/28573 was merged in too soon, and the changes should have gone in there.

Reply:
Of course (that's why I opened this PR).